### PR TITLE
feat(analytics): add live updates with play/pause button

### PR DESCRIPTION
**Pull request by yquansah**
Add live updates with play/pause button.

updates

### CODE REVIEW

**Comment by erka** [comment]
It looks great @yquansah! Should we have timestamps with 00 seconds? I am worry that those extra seconds could impact the values of the whole chart.

File: ui/src/app/flags/analytics/Analytics.tsx [hunk 1]
```diff
@@ -61,26 +62,33 @@ const durations: FilterableDurations[] = [
...
+        timeFormat
       ),
-      timeFormat
-    ),
-    to: format(addMinutes(nowISO, nowISO.getTimezoneOffset()), timeFormat)
-  });
+      to: format(addMinutes(nowISO, nowISO.getTimezoneOffset()), timeFormat)
+    },
+    {
+      pollingInterval
```
Comment (markphelps) [inline ui/src/app/flags/analytics/Analytics.tsx:89]: how does this work?

**Review by markphelps — APPROVED** [review]
one minor change request and a question

File: ui/src/app/flags/analytics/Analytics.tsx (hunk 1 above)
Comment (erka) [reply ui/src/app/flags/analytics/Analytics.tsx:89]: https://redux-toolkit.js.org/rtk-query/usage/polling

Comment (yquansah) [reply ui/src/app/flags/analytics/Analytics.tsx:89]: @markphelps Yeah I was about to point to those docs as well. I followed the example there.

Comment (markphelps) [reply ui/src/app/flags/analytics/Analytics.tsx:89]: nice!

**Comment by yquansah** [comment]
@erka Ahh i see, so what you are saying is to fix the values on the X axis to be on the minute instead of showing second granularity?

**Comment by erka** [comment]
> he values on the X axis to be on the minute in

Yes, something like https://play.grafana.org/d/000000012/grafana-play-home?orgId=1&viewPanel=5 

And another thing the api request could have different seconds between re-requests which could return different values in chart. Like time 00:01:01 may have 10 as value, but 00:01:30 could have 1 as value. Depending on polling time it will vary values in my theory.

**Comment by erka** [comment]
Those seconds lead to these result in my local env.
 ```
{
    "timestamps": [
        "2024-02-09T21:55:48Z",
        "2024-02-09T21:56:00Z",
        "2024-02-09T21:56:48Z",
        "2024-02-09T21:57:00Z",
        "2024-02-09T21:57:48Z"
....
```

If I do 
```
 const d = new Date();
  d.setSeconds(0);
  const nowISO = parseISO(d.toISOString());
```
I don't have those double minutes

**Comment by yquansah** [comment]
@erka I fixed this in the next commit, let me know if is what you are referring to.

**Comment by erka** [comment]
@yquansah check the timestamps in my debugger. Currently I see twice more timestamps that it should have

https://github.com/flipt-io/flipt/assets/19472/91d77ebc-89b9-40ff-b306-bd85d56ded5f

**Comment by erka** [comment]
@yquansah the problem is related to sql query which select data with `toStartOfInterval` and fill data with seconds. As result I got 60 records from query instead of 30.  And those with seconds have zero values.
image

**Comment by yquansah** [comment]
@erka Ah, this is a great catch. I believe the way we could fix this is to normalize the intervals to be fixed onto the previous minute.

Basically as you've done above.

**Comment by yquansah** [comment]
I wonder if this should be done on the backend as well. Probably.

**Comment by yquansah** [comment]
@erka Thank you for that!! 🙏

**Comment by yquansah** [comment]
@erka Hmm i think there is another possible issue with just setting the seconds to 0. You would have to wait a whole minute to see live updates.

We are basically duplicating the query for an entire minute before we move onto the next window. If that makes sense?

**Comment by erka** [comment]
> @erka Hmm i think there is another possible issue with just setting the seconds to 0. You would have to wait a whole minute to see live updates.
> 
> We are basically duplicating the query for an entire minute before we move onto the next window. If that makes sense?

You have to use next minute in fact.  Like add extra seconds (60 - current seconds).

But problem is more complex. That will happen if you change interval to bigger value?  like 4 hour or 1 day.  Probably all dates should be use `toStartOfInterval` in sql query, right?

**Comment by yquansah** [comment]
@erka This is great! I tried to see if it could be done with [date_trunc](https://clickhouse.com/docs/en/sql-reference/functions/date-time-functions#date_trunc), but not possible because it does not respect custom intervals. `toStartOfInterval` worked correctly though.

**Pull request merged**